tests/behaviour/util.rs

```diff
@@ -19,8 +19,158 @@
+                .unwrap()
+                .into_iter()
+                .filter(|attr| attr.type_.label == identifiers[0])
+                .collect::<Vec<_>>()
+                .first()
+                .and_then(|attr| Some(value_equals_str(&attr.value, identifiers[1])));
+            match equals {
+                Some(val) => val,
+                None => false,
+            }
+        }
+        Err(_) => false,
+    }
+}
+
+fn labels_equal(expected_label: &str, answer: &Concept) -> bool {
+    answer.type_label_cloned() == expected_label
+}
+
+fn values_equal(expected_label_and_value: &str, answer: &Concept) -> bool {
+    let identifiers: Vec<&str> = expected_label_and_value.splitn(2, ":").collect();
+    assert_eq!(identifiers.len(), 2, "Unexpected table cell format: {expected_label_and_value}.");
+    let Concept::Attribute(Attribute { value, .. }) = answer else { unreachable!() };
+    value_equals_str(value, identifiers[1])
+}
+
+fn value_equals_str(value: &Value, expected: &str) -> bool {
+    match value {
+        Value::String(val) => val == expected,
+        Value::Long(val) => expected.parse::<i64>().and_then(|expected| Ok(expected.eq(val))).unwrap_or_else(|_| false),
+        Value::Double(val) => expected
+            .parse::<f64>()
+            .and_then(|expected| Ok(equals_approximate(expected, *val)))
+            .unwrap_or_else(|_| false),
+        Value::Boolean(val) => {
+            expected.parse::<bool>().and_then(|expected| Ok(expected.eq(val))).unwrap_or_else(|_| false)
+        }
+        Value::DateTime(val) => format_datetime(val) == expected,
+    }
+}
+
+pub fn equals_approximate(first: f64, second: f64) -> bool {
+    const EPS: f64 = 1e-4;
+    return (first - second).abs() < EPS;
+}
+
+fn format_datetime(datetime: &NaiveDateTime) -> String {
+    if datetime.time() == NaiveTime::from_hms_opt(0, 0, 0).unwrap() {
+        format!("{0}", datetime.date())
+    } else {
+        format!("{0}", datetime)
+    }
+}
+
+pub async fn match_templated_answer(
+    context: &mut Context,
+    step: &Step,
+    answer: &ConceptMap,
+) -> TypeDBResult<Vec<ConceptMap>> {
+    let query = apply_query_template(step.docstring().unwrap(), answer);
+    let parsed = parse_query(&query)?;
+    Ok(context.transaction().query().match_(&parsed.to_string())?.try_collect::<Vec<_>>().await?)
+}
+
+fn apply_query_template(query_template: &String, answer: &ConceptMap) -> String {
+    let re = Regex::new(r"<answer\.(.+?)\.iid>").unwrap();
+    re.replace_all(query_template, |caps: &Captures| format!("{}", get_iid(&answer.map.get(&caps[1]).unwrap())))
+        .to_string()
+}
+
+fn get_iid(concept: &Concept) -> String {
+    let iid = match concept {
+        Concept::Entity(Entity { iid, .. }) => iid,
+        Concept::Attribute(Attribute { iid, .. }) => iid,
+        Concept::Relation(Relation { iid, .. }) => iid,
+        _ => unreachable!("Unexpected Concept type: {concept:?}"),
+    };
+    format!("0x{iid}")
```

Comment:
Does java manually format the IID like this also? Or is it built into a concept somewhere?

Reply:
Java stores `IID` as a `String`.

Reply:
Prefix has been moved to a `fmt::Display for ID`.